src/middleware/validate.js

```diff
@@ -0,0 +1,36 @@
+const validator = require('../utils/validate-helpers')
+
+
+const validateSignupData = async (req, res, next) => {
+    const validationRules = {
+        "firstname":"required|string|min:2|max:100",
+        "lastname":"required|string|min:2|max:100",
+        "email":"required|string|email",
+        "password":"required|string|min:6"
+    };
+
+    await validator(req.body, validationRules, {}, (err, status)=>{
```

Comment:
can't you see that you are rewriting this particular function multiple times, have one function for this then pass the validation rule and the req.body to it